src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Binders/BodyModelBinder.cs

```diff
@@ -21,8 +22,8 @@ public class BodyModelBinder : IModelBinder
     {
         private readonly IList<IInputFormatter> _formatters;
         private readonly Func<Stream, Encoding, TextReader> _readerFactory;
-        private readonly ILogger _logger;
         private readonly MvcOptions _options;
+        protected readonly ILogger logger;
```

Comment:
What subclasses of this binder are we supporting? Nothing else in this class is `protected` or `virtual`.

If needed, expose a `protected` get-only property instead.

Reply:
Still not seeing the point of exposing these properties. `public` does not mean "something for which we expect subclassing to be needed.

Reply:
One alternative to share the logger instances between the two types would be to have the ctor accept an `ILogger<BodyModelBinder>`. Derived types could then pass an inject `ILogger<DerivedBodyModelBinder>` and cache the instance for their own use if necessary.